### DISCUSSION TITLE
How can I mutate the store?

### DISCUSSION THREAD

**Discussion by adichannnnnhere64** [simonwep/viselect#222] [General]
I'm using react wrapper

```javascript
  const onMove = ({
    store: {
      changed: { added, removed},
    },
  }: SelectionEvent) => {

    setDragRow((prev) => {
      const next = [...prev];
      extractIds(added).forEach(id => next.push(id));
      extractIds(removed).forEach(id => {
        const index = next.indexOf(id);
          next.splice(index, 1);
      });

      return [...new Set(next)];
    });
  };
```

```javascript
const onSelectChange = (
    newSelectedRowKeys: React.Key[],
    record: any,
    action: any,
  ) => {
    setSelectedRowKeys(newSelectedRowKeys);
  };
```

I want to mutate the value of  `added` like remove a specific element selection.

store: {
      changed: { added, removed},
},

I have an onSelecteChange event that needs to mutate the current store.

**Comment by simonwep** [comment]
Each event comes with the [current instance](https://github.com/simonwep/selection/blob/master/packages/vanilla/README.md#event-properties), you should be able to use one of the [functions](https://github.com/simonwep/selection/tree/master/packages/vanilla#functions) (e.g. `deselect`) to change the selection.
